darity_test.go

```diff
@@ -1,13 +1,45 @@
+// +build linux
+
+// These tests use a mocked KVM virtual device and allow different ioctl
+// implementations to be used.
+
 package darity
 
 import (
+	"io/ioutil"
+	"os"
 	"testing"
 )
 
-func TestAPIVersion(t *testing.T) {
-	v, err := APIVersion()
+// TestAPIVersion verifies that Client.APIVersion returns the
+// KVM API version identified by the Version constant.
+func TestAPIVersionKVM(t *testing.T) {
+	c := &Client{
+		ioctl: func(fd uintptr, request int, argp uintptr) (int, error) {
+			return Version, nil
+		},
+	}
+
+	v, err := c.APIVersion()
 	if err != nil {
 		t.Errorf("could not get API version: %q", err.Error())
 	}
-	t.Logf("APIVersion: %d", v)
+
+	if want, got := v, Version; want != got {
+		t.Fatalf("unexpected KVM API version: %d != %d", want, got)
+	}
+}
+
+// tempFile creates a temporary file for use as a mock KVM virtual device, and
+// returns the file and a function to clean it up and remove it on completion.
+func tempFile(t *testing.T) (*os.File, func()) {
```

Comment:
Yep, I wrote this and ended up not needing it, but it could be handy if we need to perform reads/writes to a mock KVM device.  It could be removed for now, but I figured it wouldn't hurt since it just exists when the tests are compiled.